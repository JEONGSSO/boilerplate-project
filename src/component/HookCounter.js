import React, {useState} from 'react';

const HookCounter = ({}) => {
   const [number, setNumber] = useState(0);

   const onIncre = () => setNumber(number + 1)
   const onDecre = () => setNumber(number - 1)

   return (
      <div>
         <h2>{number}</h2>
         <button className={'pb'} onClick={onIncre}>+1</button>
         <button id={'mb'} onClick={onDecre}>-1</button>
      </div>
   )
}

export default HookCounter;