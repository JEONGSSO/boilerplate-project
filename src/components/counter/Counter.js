import React, { useState } from 'react';

export default () => {
  const [count, setCount] = useState(0);

  const increment = () => {
    setCount(count + 1);
  };

  const decrement = () => {
    setCount(count - 1);
  };

  return (
    <div>
      <button onClick={decrement}>-</button>
      <label data-testid="value">{count}</label>
      <button onClick={increment}>+</button>
    </div>
  );
};
