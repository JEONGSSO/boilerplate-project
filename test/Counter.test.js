import React from 'react';
import { shallow, mount, render  } from 'enzyme'
import HookCounter from '../src/component/HookCounter'
import Counter from '../src/component/Counter'

const arrayInMaxValue = (arr) => Math.max(...arr)
const arrayInAverageValue = (arr) => 
   arr.reduce((acc, current, { length }) =>
      acc + current / length
   , 0)



describe('카운터 테스트', () => {
   // it('스냅샷 매치', () => {
   //    const wrapper = shallow(<HookCounter />)
   //    expect(wrapper).toMatchSnapshot();
   // });
   // it('초기 값', () =>  {
   //    const wrapper = shallow(<Counter />)
   //    expect(wrapper.find('h2').text()).toBe('0')
   // })
   // it('증가', () => {
   //    const { container } = render(<HookCounter />)
   //    // const incrementButton = container.firstChild
   //    // expect(incrementButton.textContent).toBe('1')
   //    // expect(wrapper.find('h2').text()).toBe('1')
   //    console.log()
   // })
   // it('감소', () => {
   //    const wrapper = mount(<HookCounter />)
   //    expect(wrapper.find('h2').text()).toBe('-1')
   // })
   // it('증가 버튼 클릭 시뮬', () => {
   //    const wrapper = mount(<HookCounter />)
   //    const plusButton = wrapper.findWhere(
   //       node => node.type() === 'button' && node.text() === '+1'
   //    )
   //    plusButton.simulate('click');
   //    expect(wrapper.state().number).toBe(1)
   // })
   // it('감소 버튼 클릭 시뮬', () => {
   //    const wrapper = mount(<HookCounter />)
   //    wrapper.find('#mb').simulate('click')
   //    const number = wrapper.find('h2').text()
   //    expect(number).toBe("-1")
   // })
   it('MaxValue In array', ()  => {
      expect(arrayInMaxValue([1,2,3,4]) / 2).toBe(2)
   })
   it('Average In array', ()  => {
      expect(arrayInAverageValue([70,80])).toBe(75)
   })



})