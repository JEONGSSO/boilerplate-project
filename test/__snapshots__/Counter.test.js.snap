// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`카운터 테스트 스냅샷 매치 1`] = `
<div>
  <h2>
    0
  </h2>
  <button!
    onClick={[Function]}
  />
  <button
    onClick={[Function]}
  />
</div>
`;
